packages/medusa/src/services/order.ts

```diff
@@ -205,53 +219,92 @@ class OrderService extends TransactionBaseService {
       order: { created_at: "DESC" },
     }
   ): Promise<[Order[], number]> {
-    const orderRepo = this.manager_.getCustomRepository(this.orderRepository_)
+    const orderRepo = this.manager_.withRepository(this.orderRepository_)
 
     let q
     if (selector.q) {
       q = selector.q
       delete selector.q
+
+      config.relations = config.relations
+        ? Array.from(
+            new Set([...config.relations, "shipping_address", "customer"])
+          )
+        : ["shipping_address", "customer"]
     }
 
-    const query = buildQuery(selector, config)
+    const query = buildQuery(selector, config) as FindManyOptions<Order>
 
     if (q) {
-      const where = query.where
+      const where = query.where as FindOptionsWhere<Order>
 
       delete where.display_id
       delete where.email
 
-      query.join = {
-        alias: "order",
-        innerJoin: {
-          shipping_address: "order.shipping_address",
-          customer: "order.customer",
+      // Inner join like constraints
+      const innerJoinLikeConstraints = {
+        customer: {
+          id: Not(IsNull()),
+        },
+        shipping_address: {
+          id: Not(IsNull()),
         },
       }
 
-      query.where = (qb): void => {
-        qb.where(where)
-
-        qb.andWhere(
-          new Brackets((qb) => {
-            qb.where(`shipping_address.first_name ILIKE :qfn`, {
-              qfn: `%${q}%`,
-            })
-              .orWhere(`order.email ILIKE :q`, { q: `%${q}%` })
-              .orWhere(`display_id::varchar(255) ILIKE :dId`, { dId: `${q}` })
-              .orWhere(`customer.first_name ILIKE :q`, { q: `%${q}%` })
-              .orWhere(`customer.last_name ILIKE :q`, { q: `%${q}%` })
-              .orWhere(`customer.phone ILIKE :q`, { q: `%${q}%` })
-          })
-        )
-      }
+      query.where = [
```

Comment:
thought: Would it make sense to collapse these where-clauses into a single one?
```
...query.where,
          ...innerJoinLikeConstraints,
          shipping_address: {
            ...innerJoinLikeConstraints.shipping_address,
            id: Not(IsNull()),
            first_name: ILike(`%${q}%`),
          },
        display_id: { ... }
        customer: { ... }
```
Or is it not possible? 🤔

Reply:
It is a or condition, having them aggregated would become a and condition

Reply:
Got it - wasn't aware. In that case, you can discard all comments related to that 😅  

Would it make sense to add a small note on that in the migration guide?

Reply:
We definitely can, but I am not sure it is a new feature though 😀 but i ll add it in order to have it as a sort of convention as part of the migration